api/lib/domain/services/authentication/oidc-authentication-service.js

```diff
@@ -66,25 +71,25 @@ class OidcAuthenticationService {
       redirect_uri: redirectUri,
     };
 
-    const response = await httpAgent.post({
+    const httpResponse = await httpAgent.post({
       url: this.tokenUrl,
       payload: querystring.stringify(data),
       headers: { 'content-type': 'application/x-www-form-urlencoded' },
       timeout: settings.partner.fetchTimeOut,
     });
 
-    if (!response.isSuccessful) {
+    if (!httpResponse.isSuccessful) {
       const message = 'Erreur lors de la récupération des tokens du partenaire.';
-      const dataToLog = httpErrorsHelper.serializeHttpErrorResponse(response, message);
+      const dataToLog = httpErrorsHelper.serializeHttpErrorResponse(httpResponse, message);
       monitoringTools.logErrorWithCorrelationIds({ message: dataToLog });
-      throw new InvalidExternalAPIResponseError(message);
+      throw new OidcInvokingTokenEndpointError(message);
```

Comment:
```suggestion
      throw new OidcExchangeCodeForTokenError(message);
```

Cela permettrait de préciser un peu plus l'étape à laquelle on se trouve dans le flow OAuth2.

Reply:
Je suis pas hyper convaincue ^^
On peut  aussi voir le flow Oauth2 comme une série d'appels apis et dans ce cas précis, l'erreur documente le fait que l'appel à la route /acces_token échoue. 
Qu'est ce que ça apport de plus de nommer l'erreur comme le nom de la méthode ? 
Pour échanger un code contre un token, tu vois un autre moyen proposé dans la doc OAuth2 que de faire un appel à la route /access_token ?

Reply:
Pas de souci, c'était une suggestion de mettre plutôt un nom d'erreur lié à une action plutôt qu'à un appel api.

Reply:
Ok ça marche